WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,18 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event_id)
+    event_id.in?(self.events.map{ |ev| ev.id })
```

Comment:
`self.events.map... ` could be written more succinctly as `self.events.map(&:id)`